mapswipe_workers/mapswipe_workers/project_types/arbitrary_geometry/project.py

```diff
@@ -16,8 +18,49 @@ def __init__(self, project_draft: dict) -> None:
 
         # set group size
         self.groupSize = project_draft["groupSize"]
-        self.inputGeometries = project_draft["inputGeometries"]
+        self.geometry = project_draft["geometry"]
+        self.inputType = project_draft["inputType"]
         self.tileServer = vars(BaseTileServer(project_draft["tileServer"]))
+        if "filter" in project_draft.keys():
+            self.filter = project_draft["filter"]
+        if "TMId" in project_draft.keys():
+            self.TMId = project_draft["TMId"]
+
+    def handle_input_type(self, raw_input_file: str):
+        """
+        Handle different input types.
+
+        Input (inputGeometries) can be:
+        'aoi_file' -> query ohsome with aoi from geometry then write
+            result to raw_input_file
+        a Link (str) -> download geojson from link and write to raw_input_file
+        a TMId -> get project info from geometry and query ohsome
+            for objects, then write to raw_input_file.
+        """
+        if not isinstance(self.geometry, str):
```

Comment:
can you add a comment here, when we expect this to happen? E.g. is this when provding a link?